pkg/api/merge.go

```diff
@@ -0,0 +1,368 @@
+package api
+
+// TODO: needs unit tests
+
+// Notes: shortcomings of the Azure-style API include:
+// 1. There is no mechanism to delete an item from a slice or map without
+// implementing a separate CRUD endpoint.
+// 2. There is no way to set a plain value to its zero value (e.g. bool to
+// false, int to 0, string to "") without storing it as a pointer in the API.
+
+// At the moment Config isn't merged.  Perhaps what we actually need is a
+// merging converter f(*ext, *int) *int.  Perhaps this would mean we could get
+// rid of some pointers in our internal type.
+
+func Merge(complete, partial *OpenShiftManagedCluster) *OpenShiftManagedCluster {
+	complete = complete.DeepCopy() // required because Config isn't merged
+
+	return mergeOpenShiftManagedCluster(complete, partial)
+}
+
+func mergeOpenShiftManagedCluster(complete, partial *OpenShiftManagedCluster) *OpenShiftManagedCluster {
+	switch {
+	case complete == nil && partial == nil:
+		return nil
+	case complete == nil:
```

Comment:
does this need to move up to `Merge` since you're calling `DeepCopy` on a possibly nil `OpenShiftManagedCluster`?

Reply:
I think the DeepCopy() just needs to be guarded with an `if complete != nil`.  Thanks, will fix.